File: src/msa/web-service/portalapi/src/main/java/vn/com/kns/portalapi/application/service/category/province/dto/ProvinceDto.java
package vn.com.kns.portalapi.application.service.category.province.dto;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import vn.com.kns.portalapi.application.service.category.country.dto.CountryDto;
import vn.com.kns.portalapi.core.model.BaseEntityDto;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class ProvinceDto extends BaseEntityDto {
    private String name;
    private CountryDto country;
    private Long countryId;

//    private ZoneDto zone;
//    private Long zoneId;
}

File: src/msa/web-service/portalweb/src/app/shared/services/category/province/dto/province-dto.ts
import {CountryDto} from '../../country/dto/country-dto';

export class ProvinceDto {
  id: number | null;
  name: string;
  countryId: number | null;
  country: CountryDto = new CountryDto();

  constructor(id?: number | null, name?: string) {
    this.id = id;
    this.name = name;
  }

}
